File: advent2021/day2.ts
import fs = require("fs");
import readline = require("readline");

async function processLineByLine() {
  const fileStream = fs.createReadStream("inputs/day2.txt");

  const rl = readline.createInterface({
    input: fileStream,
    crlfDelay: Infinity,
  });

  let data: string[][] = [[]];
  for await (const line of rl) {
    data.push(line.split(" "));
  }

  let horizontal: number = 0;
  let depth: number = 0;

  data.forEach((i) => {
    switch (i[0]) {
      case "forward":
        horizontal += Number(i[1]);
        break;
      case "up":
        depth -= Number(i[1]);
        break;
      case "down":
        depth += Number(i[1]);
        break;
      default:
        break;
    }
  });

  console.log(horizontal * depth);
}

async function processLineByLine2() {
  const fileStream = fs.createReadStream("inputs/day2.txt");

  const rl = readline.createInterface({
    input: fileStream,
    crlfDelay: Infinity,
  });

  let data: string[][] = [[]];
  for await (const line of rl) {
    data.push(line.split(" "));
  }

  let horizontal: number = 0;
  let depth: number = 0;
  let aim: number = 0;

  data.forEach((i) => {
    switch (i[0]) {
      case "forward":
        horizontal += Number(i[1]);
        depth += aim * Number(i[1]);
        break;
      case "up":
        aim -= Number(i[1]);
        break;
      case "down":
        aim += Number(i[1]);
        break;
      default:
        break;
    }
  });

  console.log(horizontal*depth);
}

// processLineByLine();
processLineByLine2();

File: advent2021/day2.py
def part1():
    file=open("inputs/day2.txt","rt")
    data=file.readlines()
    h=0
    d=0
    for i in range(len(data)):
        fields=data[i].split()
        if fields[0] == 'forward':
            h+=int(fields[1])
        elif fields[0] == 'up':
            d-=int(fields[1])
        elif fields[0] == 'down':
            d+=int(fields[1])
    print(h*d)

def part2():
    file=open("inputs/day2.txt","rt")
    data=file.readlines()
    h=0
    d=0
    a=0
    for i in range(len(data)):
        fields=data[i].split()
        if fields[0] == 'forward':
            h+=int(fields[1])
            d+=a*int(fields[1])
        elif fields[0] == 'up':
            a-=int(fields[1])
        elif fields[0] == 'down':
            a+=int(fields[1])
    print(h*d)

part1()
part2()